src/main/java/com/es/phoneshop/model/cart/service/DefaultCartService.java

```diff
@@ -61,21 +56,26 @@ public void add(HttpSession currentSession, Cart cart, Long productId, int quant
         }
     }
 
-    private int getAvailableProductQuantityToAddToCart(CartItem cartItem, Product product) {
-        if (cartItem == null) {
-            return product.getStock();
-        }
-        int currentProductCartQuantity = cartItem.getQuantity();
-        int productStock = cartItem.getProduct().getStock();
-        return productStock - currentProductCartQuantity;
+    private int getAvailableProductQuantity(Optional<CartItem> cartItem, Product product) {
+        return cartItem.map(currentCartItem -> {
+            int currentProductCartQuantity = currentCartItem.getQuantity();
+            int productStock = currentCartItem.getProduct().getStock();
+            return productStock - currentProductCartQuantity;
+        }).orElse(product.getStock());
     }
 
-    private void addProductToCart(Cart cart, CartItem cartItem, Product product, int quantity) {
-        if (cartItem != null) {
-            cartItem.setQuantity(cartItem.getQuantity() + quantity);
-        } else {
-            cart.getItems().add(new CartItem(product, quantity));
-        }
+    private void addProductToCart(Cart cart, Optional<CartItem> cartItem, Product product, int quantity) {
```

Comment:
i don't recommend to use Optional as method argument:
https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments

Reply:
you can leave part of this logic (whether cartItem is present or not) in the add() method

Reply:
you can leave part of this logic (whether cartItem is present or not) in the add() method